### ISSUE TITLE
Documentation about Networking Options missing

### ISSUE THREAD

**Issue by uwedisch** [e2guardian/e2guardian#659]
It would be very helpful to have some documentation about the networking options someone has with E2Guardian.

Before using E2Guardian and doing system design some questions arise and for example I found no answer to the next ones:
- Option for incoming (bind/listen) interface
- Option for outgoing interface (is it possible to use other interface for internet traffic than default route?)
- Can E2Guardian work as a HTTP/HTTPS or a SOCKS proxy or both beneath transparent mode?

To sum it up: A documentation of all configuration variables related to networking is missing.

**Comment by philipianpearce** [comment]
Which version are you looking at?

A lot of work has been done on documentation in v5.4 and this work is continuing with both 5.4 and 5.5.

There is a NETWORK section in the example e2guardian.conf which describes all the options, including those for incoming ip/ports.  https://github.com/e2guardian/e2guardian/blob/v5.4.dev/configs/e2guardian.conf.in

"Option for incoming (bind/listen) interface
Option for outgoing interface (is it possible to use other interface for internet traffic than default route?)"

Which interface is used is configured in the operating system - it is not configurable within an application as applications such as e2g communicate with the o/s at level 3, i.e. at IP layer.   So look at your platform documentation for how to do this.

"Can E2Guardian work as a HTTP/HTTPS or a SOCKS proxy or both beneath transparent mode?"
I'm not clear what you are asking here.

E2G is a http/https filtering proxy and supports both explicit and transparent proxy modes at the same time.  To enable http transparent mode, o/s firewall port redirection is required.   Https transparent mode also requires o/s firewall redirection as well as https port being set using transparenthttpsport - all as documented in e2guardian.conf.

For transparent mode the system running e2g must be the default gateway for the users.
Other proxies such as SOCKS could also be run on the hosting system, but this is outside scope for e2g documentation.

**Comment by uwedisch** [comment]
OK, thank you.

So, it seems that an option similar to `tcp_outgoing_address` of Squid or `Bind` of Tinyproxy is not implemented.

**Comment by philipianpearce** [comment]
That is correct.    If it is required then I will add to the 'wishlist' for future versions.
